OpenCV 3.3.0 IOS Dynamic Libraries Build Error IPPICV Undefined Symbols for iPhoneSimulator x86_64 Arch

##### System information
- OpenCV => 3.3.0
- Operating System / Platform => MacOS High Sierra v10.13
- Hardware => MacBook Pro (Retina, 13-inch, Early 2015), Intel Iris Graphics 6100 1536 MB 
- Compiler => LLVM/Clang(++) 
- C/CXX Compiler => AppleClang
- CMAKE Version => cmake version 3.9.4
- Python Interpreter => Python Interpreter for `build_framework.py` installed via homebrew v3.6. 

##### Detailed description

I am having a problem building the iOS dynamic libraries for iPhoneSimulator x86_64, via the `build_framework.py` script, in the `platforms/ios` directory. I am using python3.6 installed via Homebrew.

There appears to be an `ld` linker error for arch iPhoneSimulator x86_64.

It seems like the problem is because of ippicv dependencies not being downloaded for the  iPhoneSimulator x86_64 arch. It is clearly missing in the 3rd Party libraries, when downloaded. However, the ippicv cmake file does appear to download the i386 libs for the iPhoneSimiulator-- on a side note, why is this even supported? Seems like it is legacy for dynamic libraries, anyways? 

Even though the 3rd party ippicv appears to be missing for iPhoneOS Simulator x86_64, there is still a reference to build for iPhoneSimulator x86_64 in `platform/ios/build_framework.py` on line 285. So, either the libs are not being downloaded correctly, or it shouldn't be supported. 

##### Steps to reproduce
* Pull opencv repository and checkout v3.3.0 tag.
* Navigate to `platforms/ios/` 
* Run `build_framework.py --dynamic ios` under the python3.6 interpreter installed via homebrew.

The exact command that fails is the following: 
* `'['xcodebuild', 'IPHONEOS_DEPLOYMENT_TARGET=8.0', 'ONLY_ACTIVE_ARCH=NO', '-arch', 'i386', '-arch', 'x86_64', '-sdk', 'iphonesimulator', '-configuration', 'Release', '-parallelizeTargets', '-jobs', '4', '-target', 'ALL_BUILD', '-target', 'ALL_BUILD', 'build']'`, specifically in the `-arch x86_64` linking.

With the following sample error, attached is the cmake summary:
* `ld: warning: ignoring file /Users/<USER>/opencv/platforms/ios/ios/build/build-iphonesimulator/3rdparty/ippicv/ippicv_mac/../ippiw_mac//lib/ia32/libipp_iw.a, file was built for archive which is not the architecture being linked (x86_64): /Users/<USER>/opencv/platforms/ios/ios/build/build-iphonesimulator/3rdparty/ippicv/ippicv_mac/../ippiw_mac//lib/ia32/libipp_iw.a
Undefined symbols for architecture x86_64:
  "_ippicviThreshold_GT_32f_C1R", referenced from:
      cv::ThresholdRunner::operator()(cv::Range const&) const in thresh.o
`

##### Notes
I am not an experienced MacOS user, so sorry if the fix is a simple one, or obvious.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -204,7 +204,7 @@ OCV_OPTION(WITH_GSTREAMER      "Include Gstreamer support"                   ON 
 OCV_OPTION(WITH_GSTREAMER_0_10 "Enable Gstreamer 0.10 support (instead of 1.x)"                              OFF )
 OCV_OPTION(WITH_GTK            "Include GTK support"                         ON   IF (UNIX AND NOT APPLE AND NOT ANDROID) )
 OCV_OPTION(WITH_GTK_2_X        "Use GTK version 2"                           OFF  IF (UNIX AND NOT APPLE AND NOT ANDROID) )
-OCV_OPTION(WITH_IPP            "Include Intel IPP support"                   NOT MINGW IF (X86_64 OR X86) AND NOT WINRT )
+OCV_OPTION(WITH_IPP            "Include Intel IPP support"                   NOT MINGW IF (X86_64 OR X86) AND NOT WINRT AND NOT IOS )
 OCV_OPTION(WITH_HALIDE         "Include Halide support"                      OFF)
 OCV_OPTION(WITH_JASPER         "Include JPEG2K support"                      ON   IF (NOT IOS) )
 OCV_OPTION(WITH_JPEG           "Include JPEG support"                        ON)

